packages/shared/test/permissions.test.ts

```diff
@@ -130,6 +132,20 @@ describe("Role permissions", () => {
     expect(p.canViewFeatureModal()).toBe(false);
     expect(p.canViewIdeaModal()).toBe(false);
     expect(p.canViewReportModal()).toBe(false);
+    expect(p.canCreateAndUpdateFactFilter(projectsResource)).toBe(false);
+    expect(p.canDeleteFactFilter(projectsResource)).toBe(false);
+    expect(p.canCreateOrUpdateEnvironment(environmentsResource)).toBe(false);
+    expect(p.canDeleteEnvironment(environmentsResource)).toBe(false);
+    expect(p.canViewCreateSDKConnectionModal(project)).toBe(false);
+    expect(p.canCreateSDKConnection({ projects, environment })).toBe(false);
+    expect(p.canUpdateSDKConnection({ projects, environment }, updates)).toBe(
+      false
+    );
+    expect(p.canDeleteSDKConnection({ projects, environment })).toBe(false);
+    expect(p.canReadSingleProjectResource(project)).toBe(false);
+    // This is a weird one - for resources that have multiple projects, if their projects array is empty
+    // that means it's in 'All Projects' so users with global noaccess can view them
+    expect(p.canReadMultiProjectResource(projects)).toBe(true);
```

Comment:
Given your comment, I would add another one here checking when projects is `["foo"]`

Reply:
I guess we could improve the permission logic for this case to require the user to have at least one project-scoped permission that has read access.

There's not a good reason to mark someone as no access globally without overriding it in a project, so it's a little artificial

Reply:
@jdorn Pushed up a commit [here](https://github.com/growthbook/growthbook/pull/2499/commits/9bb52295f268189ddda9e6622441b3d9fbc6aa62) that updates the logic based on the above convo. Could probably still use some optimization.